tests/test_numpy_wavefunction.py

```diff
@@ -20,106 +20,6 @@
-        expected = np.kron(np.kron(U2, I), I)
-        self.assertAllAlmostEqual(res, expected, delta=tol)
-
-        # test applied to wire 1+2
-        res = dev.expand(U2, [1, 2])
-        expected = np.kron(np.kron(I, U2), I)
-        self.assertAllAlmostEqual(res, expected, delta=tol)
-
-        # test applied to wire 2+3
-        res = dev.expand(U2, [2, 3])
-        expected = np.kron(np.kron(I, I), U2)
-        self.assertAllAlmostEqual(res, expected, delta=tol)
-
-        # CNOT with target on wire 1
-        res = dev.expand(CNOT, [1, 0])
-        rows = np.array([0, 2, 1, 3])
-        expected = np.kron(np.kron(CNOT[:, rows][rows], I), I)
-        self.assertAllAlmostEqual(res, expected, delta=tol)
-
-        # test exception raised if unphysical subsystems provided
-        with pytest.raises(
-            ValueError, match="Invalid target subsystems provided in 'wires' argument."
-        ):
-            dev.expand(U2, [-1, 5])
-
-        # test exception raised if incorrect sized matrix provided
-        with pytest.raises(ValueError, match="Matrix parameter must be of size"):
-            dev.expand(U, [0, 1])
-
-    def test_expand_three(self, tol):
-        """Test that a 3 qubit gate correctly expands to 4 qubits."""
-        dev = plf.NumpyWavefunctionDevice(wires=4)
-
-        # test applied to wire 0,1,2
-        res = dev.expand(U_toffoli, [0, 1, 2])
-        expected = np.kron(U_toffoli, I)
-        self.assertAllAlmostEqual(res, expected, delta=tol)
-
-        # test applied to wire 1,2,3
-        res = dev.expand(U_toffoli, [1, 2, 3])
-        expected = np.kron(I, U_toffoli)
-        self.assertAllAlmostEqual(res, expected, delta=tol)
-
-        # test applied to wire 0,2,3
-        res = dev.expand(U_toffoli, [0, 2, 3])
-        expected = (
-            np.kron(SWAP, np.kron(I, I)) @ np.kron(I, U_toffoli) @ np.kron(SWAP, np.kron(I, I))
-        )
-        self.assertAllAlmostEqual(res, expected, delta=tol)
-
-        # test applied to wire 0,1,3
-        res = dev.expand(U_toffoli, [0, 1, 3])
-        expected = (
-            np.kron(np.kron(I, I), SWAP) @ np.kron(U_toffoli, I) @ np.kron(np.kron(I, I), SWAP)
-        )
-        self.assertAllAlmostEqual(res, expected, delta=tol)
-
-        # test applied to wire 3, 1, 2
-        res = dev.expand(U_toffoli, [3, 1, 2])
-        # change the control qubit on the Toffoli gate
-        rows = np.array([0, 4, 1, 5, 2, 6, 3, 7])
-        expected = np.kron(I, U_toffoli[:, rows][rows])
-        self.assertAllAlmostEqual(res, expected, delta=tol)
-
-        # test applied to wire 3, 0, 2
-        res = dev.expand(U_toffoli, [3, 0, 2])
-        # change the control qubit on the Toffoli gate
-        rows = np.array([0, 4, 1, 5, 2, 6, 3, 7])
-        expected = (
-            np.kron(SWAP, np.kron(I, I))
-            @ np.kron(I, U_toffoli[:, rows][rows])
-            @ np.kron(SWAP, np.kron(I, I))
-        )
-        self.assertAllAlmostEqual(res, expected, delta=tol)
-
```

Comment:
:+1: I'll add https://github.com/rigetti/pennylane-forest/pull/17#discussion_r308577580 as a test

Reply:
How about choosing the name `test_device.py` instead of `test_mat_vec_product.py`? That way, any future common methods could also be tested in the same file, and it also satisfies the similar pattern of `test_xx` where `xx` is some sort of simulator/device (with the exception of `test_gradients.py` perhaps)

Reply:
Good call, I've just made the change (`test_device.py`) as you suggested